tests/js/assignment/destructuring-heuristic.js

```diff
@@ -0,0 +1,28 @@
+{{
+  const {
+    id,
+    static: isStatic,
+    method: isMethod,
+    methodId,
+    getId,
+    setId,
+  } = privateNamesMap.get(name);
+
+  const {
+    id1,
+    method: isMethod1,
+    methodId1
+  } = privateNamesMap.get(name);
+
+  const {
+    id2,
+    method: isMethod2,
+    methodId2
+  } = privateNamesMap.get(bifornCringerMoshedPerplexSawder);
```

Comment:
Sorry @thorn0, I think it is **this** one that is now changed since 2.2.1.

## Prettier 2.2.1

**Prettier 2.2.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEACVkoGcauAHSnXQEsATAJgBpDjUBbOGACwjKVRKwFknXKaRYoxZsAkpVqoAvqgC8qAA4AnEgDcAhvAByGxjw2KAdAHMmACgBGJAGYRlUAMKqoZ5dwhZmcMgAU4yooANnAAHgDKGgDuZAEAlADcIFQgEIowJNBYyKAaysoQUb55CNkoGkFRGgCe2SmWyhpgANZM4YpNJK7IMMoArnApcPSWPrFkADIarn0aZgBi9vRaGd3lfTAQySDMMPRBAOrMJPBYHWBw4aUn6ifVyOBYdSBdWAEwvo0my8g2FW8pABWWFCACFGi02no4BMunBfv9BiBgREuiYQgBFPoQeAIoIAkAdZRvZQPSwaUZBbYqLowA7kFjIAAcAAYUipPHADo1FA8VHASWp4SkAI7Y+CfNJlEAaLAAWigcB8Pm2yjgYpIas+cx+SD++KRb3oJB6-UNaMx4vhesRKRgFPpZEZSGoIF6GhIQTRjgg9F1IAFAFZtn03gAVCllfUEtQDMRQWKwcJgVTpACCCfCMGqITxb2k0iAA)
<!-- prettier-ignore -->
```sh
--parser babel
```

**Input:**
<!-- prettier-ignore -->
```jsx
  const {
    id2,
    method: isMethod2,
    methodId2
  } = privateNamesMap.get(bifornCringerMoshedPerplexSawder);
```

**Output:**
<!-- prettier-ignore -->
```jsx
const { id2, method: isMethod2, methodId2 } = privateNamesMap.get(
  bifornCringerMoshedPerplexSawder
);

```

## Prettier main as of recent PR 10416

same as 2.2.1 [here](https://deploy-preview-10416--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEACVkoGcauAHSnXQEsATAJgBpDjUBbOGACwjKVRKwFknXKaRYoxZsAkpVqoAvqgC8qAA4AnEgDcAhvAByGxjw2KAdAHMmACgBGJAGYRlUAMKqoZ5dwhZmcMgAU4yooANnAAHgDKGgDuZAEAlADcIFQgEIowJNBYyKAaysoQUb55CNkoGkFRGgCe2SmWyhpgANZM4YpNJK7IMMoArnApcPSWPrFkADIarn0aZgBi9vRaGd3lfTAQySDMMPRBAOrMJPBYHWBw4aUn6ifVyOBYdSBdWAEwvo0my8g2FW8pABWWFCACFGi02no4BMunBfv9BiBgREuiYQgBFPoQeAIoIAkAdZRvZQPSwaUZBbYqLowA7kFjIAAcAAYUipPHADo1FA8VHASWp4SkAI7Y+CfNJlEAaLAAWigcB8Pm2yjgYpIas+cx+SD++KRb3oJB6-UNaMx4vhesRKRgFPpZEZSGoIF6GhIQTRjgg9F1IAFAFZtn03gAVCllfUEtQDMRQWKwcJgVTpACCCfCMGqITxb2k0iAA)

## PR 10643 (this PR)

**Prettier pr-10643**
[Playground link](https://deploy-preview-10643--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEACVkoGcauAHSnXQEsATAJgBpDjUBbOGACwjKVRKwFknXKaRYoxZsAkpVqoAvqgC8qAA4AnEgDcAhvAByGxjw2KAdAHMmACgBGJAGYRlUAMKqoZ5dwhZmcMgAU4yooANnAAHgDKGgDuZAEAlADcIFQgEIowJNBYyKAaysoQUb55CNkoGkFRGgCe2SmWyhpgANZM4YpNJK7IMMoArnApcPSWPrFkADIarn0aZgBi9vRaGd3lfTAQySDMMPRBAOrMJPBYHWBw4aUn6ifVyOBYdSBdWAEwvo0my8g2FW8pABWWFCACFGi02no4BMunBfv9BiBgREuiYQgBFPoQeAIoIAkAdZRvZQPSwaUZBbYqLowA7kFjIAAcAAYUipPHADo1FA8VHASWp4SkAI7Y+CfNJlEAaLAAWigcB8Pm2yjgYpIas+cx+SD++KRb3oJB6-UNaMx4vhesRKRgFPpZEZSGoIF6GhIQTRjgg9F1IAFAFZtn03gAVCllfUEtQDMRQWKwcJgVTpACCCfCMGqITxb2k0iAA)
<!-- prettier-ignore -->
```sh
--parser babel
```

**Input:**
<!-- prettier-ignore -->
```jsx
  const {
    id2,
    method: isMethod2,
    methodId2
  } = privateNamesMap.get(bifornCringerMoshedPerplexSawder);
```

**Output:**
<!-- prettier-ignore -->
```jsx
const {
  id2,
  method: isMethod2,
  methodId2,
} = privateNamesMap.get(bifornCringerMoshedPerplexSawder);

```

---

I would favor adding this to the CHANGELOG entry.

Reply:
#10222 caused a lot of changes. We can't add all of them to the changelog.